src/common.jl

```diff
@@ -47,3 +47,20 @@ end
 
 # Tracker's implementation of ldiv isn't good. We'll use Zygote's instead.
 zygote_ldiv(A::AbstractMatrix, B::AbstractVecOrMat) = A \ B
+
+function randnsimilar(rng, x::AbstractArray, dims...)
+    randn!(rng, similar(x, dims...))
+end
+
+function randnsimilar(rng, x::CuArray, dims...)
+    adapt(CuArray, randn(rng, eltype(x), dims...))
+end
+
+# TODO: should be replace by @non_differentiable when
+# https://github.com/JuliaDiff/ChainRulesCore.jl/issues/150 is fixed
+function ChainRules.rrule(::typeof(randnsimilar), rng, x, dims...)
+    function randnsimilar_pullback(ΔQ)
+        return (NO_FIELDS, Zero(), Zero(), map(_ -> Zero(), dims)...)
+    end
+    randnsimilar(rng, x, dims...), randnsimilar_pullback
+end
```

Comment:
```suggestion
ChainRulesCore.@non_differentiable randnsimilar
```
We should also adjust any existing lower bound on ChainRulesCore to the version in which it was introduced.

Reply:
I think `@non_differentiable` should be used with types, but in this case we have the `dims...` vararg, which is not supported by the macro yet (https://github.com/JuliaDiff/ChainRulesCore.jl/issues/212).

Reply:
Ah good catch!

Reply:
But I will add an updated note :)